### PR TITLE
#r and #load completion

### PR DESCRIPTION
**Pull request by vasily-kirichenko**

### CODE REVIEW

**Comment by smoothdeveloper** [comment]
that looks awemazing, does it work with / ?

File: vsintegration/src/FSharp.Editor/Completion/ReferenceDirectiveCompletionProvider.fs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            
+            let fileSystemHelper = 
+                FileSystemCompletionHelper(
+                    this, 
+                    textChangeSpan,
+                    getFileSystemDiscoveryService snapshot,
+                    Glyph.OpenFolder,
+                    Glyph.Assembly,
+                    searchPaths = ImmutableArray.Create ".",
+                    allowableExtensions = [|".dll"; ".exe" |],
```
Comment (smoothdeveloper) [inline vsintegration/src/FSharp.Editor/Completion/ReferenceDirectiveCompletionProvider.fs]: can you consider make that work for .fsx / .fs in #load scripts as well?

**Comment by matthid** [comment]
> that looks awemazing, does it work with / ?

Good point. Thinking about it:
Maybe we should even consider to not support `\` in the tooling to push people into using `/` for cross compat reasons?

**Comment by forki** [comment]
we should support \ but maybe add / when things are done automagically.

**Comment by isaacabraham** [comment]
This is brilliant.

**Comment by matthid** [comment]
> we should support \ but maybe add / when things are done automagically.

Ideally one is able to press Strg+Z directly after this to undo (just in case one really wants to write/use `\`. Just like R#/Visual Studio works on such situations

**Comment by vasily-kirichenko** [comment]
@isaacabraham

**Comment by isaacabraham** [comment]
Even more betterer stuff :) How will this work with e.g. Ionide?

**Comment by vasily-kirichenko** [comment]
Why?..

**Comment by smoothdeveloper** [comment]
We will port his excellent code to Fsharp-Editing

**Comment by vasily-kirichenko** [comment]
It's ready.

**Comment by smoothdeveloper** [comment]
@vasily-kirichenko could you add in comments at the top of the module or next to relevant code which place you used as inspiration from roslyn codebase?

The code looks nice approachable with those few comments explaining the implementation details :+1:

**Comment by saul** [comment]
Looks great :) 

Any chance we could add icons for the script files and the DLLs? Just use the same icons that are in the solution explorer.

**Comment by saul** [comment]
This issue closes #2343

**Comment by KevinRansom** [comment]
@vasily-kirichenko 
Thank you for this, it looks good.

**Comment by vasily-kirichenko** [comment]
> @saul Any chance we could add icons for the script files and the DLLs?
For dll/exe - yes, they showed, but I lost them during refactoring, will fix in a separate PR. About fs and fsx - no, there are no icons in Roslyn for them (C#/VB only)

**Pull request merged**